### PR TITLE
Load `/directory/room/{roomAlias}` endpoint on workers

### PR DESCRIPTION
**Pull request by dklimpel**
Fixes: #13516
- #13516

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

Signed-off-by: Dirk Klimpel [5740567+dklimpel@users.noreply.github.com](mailto:5740567+dklimpel@users.noreply.github.com)

### CODE REVIEW

**Comment by dklimpel** [comment]
Sytest seems to fail https://github.com/matrix-org/sytest/actions/runs/4535947435/jobs/7992024284?pr=1343

**Comment by dklimpel** [comment]
Looks like the matching does not work

```
--2023-03-27 19:44:39--  https://github.com/matrix-org/synapse/archive/patch-1.tar.gz
Resolving github.com (github.com)... 140.82.114.3
Connecting to github.com (github.com)|140.82.114.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://codeload.github.com/matrix-org/synapse/tar.gz/patch-1 [following]
--20[23](https://github.com/matrix-org/sytest/actions/runs/4535947435/jobs/7992024284?pr=1343#step:4:24)-03-27 19:44:39--  https://codeload.github.com/matrix-org/synapse/tar.gz/patch-1
Resolving codeload.github.com (codeload.github.com)... 140.82.112.9
Connecting to codeload.github.com (codeload.github.com)|140.82.112.9|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2023-03-27 19:44:39 ERROR 404: Not Found.


gzip: stdin: unexpected end of file
tar: Child returned status 1
tar: Error is not recoverable: exiting now
--2023-03-27 19:44:39--  https://github.com/matrix-org/synapse/archive/develop.tar.gz
Resolving github.com (github.com)... 140.82.114.3
Connecting to github.com (github.com)|140.82.114.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://codeload.github.com/matrix-org/synapse/tar.gz/refs/heads/develop [following]
--2023-03-[27](https://github.com/matrix-org/sytest/actions/runs/4535947435/jobs/7992024284?pr=1343#step:4:28) 19:44:[39](https://github.com/matrix-org/sytest/actions/runs/4535947435/jobs/7992024284?pr=1343#step:4:40)--  https://codeload.github.com/matrix-org/synapse/tar.gz/refs/heads/develop
Resolving codeload.github.com (codeload.github.com)... 1[40](https://github.com/matrix-org/sytest/actions/runs/4535947435/jobs/7992024284?pr=1343#step:4:41).82.112.9
Connecting to codeload.github.com (codeload.github.com)|140.82.112.9|:4[43](https://github.com/matrix-org/sytest/actions/runs/4535947435/jobs/7992024284?pr=1343#step:4:44)... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/x-gzip]
```

File: synapse/rest/__init__.py [hunk 1]
```diff
@@ -100,8 +100,7 @@ def register_servlets(client_resource: HttpServer, hs: "HomeServer") -> None:
         login.register_servlets(hs, client_resource)
         profile.register_servlets(hs, client_resource)
         presence.register_servlets(hs, client_resource)
-        if is_main_process:
-            directory.register_servlets(hs, client_resource)
+        directory.register_servlets(hs, client_resource)
```
Comment (reivilibre) [inline synapse/rest/__init__.py:103]: This is also loading `ClientAppserviceDirectoryListServer` and `ClientDirectoryListServer` on the workers.

At a skim, it looks fine, but it should then have a `CATEGORY` and it should be in the path map like the other endpoints?
If you don't have time to check that they're fine on workers, I'd add `if is_main_process` inside `directory.register_servlets` and leave those until later.

**Review by reivilibre — CHANGES_REQUESTED** [review]
looks fine otherwise, `ClientDirectoryServer` seemed valid

File: synapse/rest/__init__.py (hunk 1 above)
Comment (dklimpel) [reply synapse/rest/__init__.py:103]: > This is also loading ClientAppserviceDirectoryListServer and ClientDirectoryListServer on the workers.

I have it disabled.

**Review by reivilibre — APPROVED** [review]

**Pull request merged**